### PR TITLE
Kr1

### PR DESCRIPTION
**Pull request by Arcimiendar**
i'd like to merge test!

### CODE REVIEW

File: main/kr1/zadacza1.py [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+import types
+import typing
+
+
+namedtuple = collections.namedtuple
+
+
+def log_decorator(func: typing.Callable, name_func: str):
+    def new_func(self, *args, **kwargs):
+
```
Comment (dmitry-mavchun) [inline main/kr1/zadacza1.py]: I think you overuse empty lines here

File: main/kr1/zadacza1.py [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+    return new_func
+
+
+@property
+def last_log(self):
+
+    self.log.reverse()
+
+    log = list()
```
Comment (dmitry-mavchun) [inline main/kr1/zadacza1.py]: `log = []`

File: main/kr1/zadacza1.py [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+
+
+class Logger(type):
+
+    @staticmethod
+    def get_new_attr(dct: typing.Dict[str, typing.Any]):
+        new_dct = dict()
+        new_dct["LogItem"] = namedtuple('LogItem', ['name', 'args', 'kwargs', 'result'],
+                                        defaults=[str(), list(), dict()])
+        new_dct["log"] = list()
```
Comment (dmitry-mavchun) [inline main/kr1/zadacza1.py]: try to keep code style persistent over project, so `new_dct['log'] = []`

File: main/kr1/zadacza2.py [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+
+
+File = typing.TextIO
+
+
+def xlsx_to_json(xlsx_workbook, json_file: File):
+
+    workbook = xlsx_workbook
+    sheet = workbook[workbook.sheetnames[0]]
+    further_json = dict()
```
Comment (dmitry-mavchun) [inline main/kr1/zadacza2.py]: `= {}`

File: main/kr1/zadacza1.py (hunk 1 above)
Comment (Arcimiendar) [reply main/kr1/zadacza1.py]: Done!

**Pull request merged**